### ISSUE TITLE
 Files uploaded with '--allow-uploads page' and '--base-path' are stored in the wrong directory

### ISSUE THREAD

**Issue by Cutkh** [gollum/gollum#1830]
**Describe the bug**
I have a gollum instance within a Docker environment that is started with the following parameters:
```
gollum --bare  --allow-uploads page --base-path myBasePath
```
When I drag&drop a file `MyFile.pdf` in the editor for the wiki page `/Projects/SomeStuff.md` it creates a link to 
`[[MyFile.pdf | /uploads/Projects/SomeStuff/MyFile.pdf]]`.
However, the file is actually saved under
`/uploads/myBasePath/gollum/edit/Projects/SomeStuff/MyFile.pdf`

The URL of the editor page is thereby:
`https://example.org/myBasePath/gollum/edit/Projects/SomeStuff.md`

**Expected behavior**
I expected the file to be saved under the path given in the generated link.

My best guess is that the base-path parameter impacts the generation of the correct path, because the actual upload-path contains `myBasePath/gollum/edit/` much like the URL of the editor page.

**Environment Info**
```
Gollum 5.3.0
Running on: x86_64-linux-musl with Ruby version 2.7.6
Using:
rugged 1.1.1
gollum-rugged_adapter 1.1.2
gollum-lib 5.2
Markdown rendering gem: kramdown
Other renderering gems:
org-ruby 0.9.12

```

Dockerfile
```
FROM ruby:2.7-alpine AS builder

RUN apk add \
    build-base \
    cmake \
    git \
    icu-dev \
    openssl-dev
                                                                                                                                                                                                                                          
RUN gem install github-linguist
RUN gem install gollum
RUN gem install org-ruby  # optional

ENV GOLLUMARGS=""

WORKDIR /wiki
EXPOSE 4567

CMD gollum $GOLLUMARGS
```
Actual docker command:
```
docker build -t gollum .

docker run --restart=unless-stopped -d -p 127.0.0.1:4567:4567  -e GOLLUMARGS="--bare  --allow-uploads page --base-path myBasePath" -v /tmp/testWiki:/wiki --name testWiki gollum

```

Cmd used to create the git directory
```
git init --bare 
```

**Comment by dometto** [comment]
Thanks for the report @Cutkh. I'm trying to reproduce this, but it actually seems as if the error is present even when not using `--base-path`. Here is the commit message I get when drag/dropping a file using `--allow-uploads page` but without `--base-path`:

```
Uploaded file to uploads/gollum/edit/Home/Screenshot 2021-10-03 at 17.43.49.png
```

**Comment by dometto** [comment]
Proposed fix in #1831
